cmd/ecs-service-logs/main.go

```diff
@@ -425,6 +446,28 @@ func showFunction(cmd *cobra.Command, args []string) error {
 	pageSize := v.GetInt(flagPageSize)
 	environment := v.GetString(flagEnvironment)
 
+	startTimeString := v.GetString(flagStartTime)
+	endTimeString := v.GetString(flagEndTime)
+	var startTimeUnix, endTimeUnix *int64
+	if len(startTimeString) > 0 && len(endTimeString) > 0 {
+		startTime, errStartTime := time.Parse(time.RFC3339, startTimeString)
+		if errStartTime != nil {
+			return errStartTime
+		}
+
+		endTime, errEndTime := time.Parse(time.RFC3339, endTimeString)
+		if errEndTime != nil {
+			return errEndTime
+		}
+
+		startTimeUnix = aws.Int64(startTime.Unix() * 1000) // milliseconds
+		endTimeUnix = aws.Int64(endTime.Unix() * 1000)     // milliseconds
+
+		if *startTimeUnix > *endTimeUnix {
+			return errors.New("the time ranges you specified do not overlap")
```

Comment:
I'd use the special error here.

```suggestion
			return &errInvalidTimeRange{StartTime: startTime, EndTime: endTime}
```